Make use of env variables free from satpy (#113)
pytroll/pyorbital: doc/source/index.rst

Pyorbital is a python package to compute orbital parameters for satellites from
TLE files as well as astronomical parameters of interest for satellite remote sensing.
Currently Pyorbital only supports low earth orbit satellites.

Installation
------------

> Pyorbital comes with a file *platforms.txt* that maps a satellite name to the NORAD identifier.

Reviewer: This section is called "Installation", but the entire section is about customizing an installation that, as it says below, should not be needed by most users as the builtin platforms file should be sufficient.

Could this section "Installation" be renamed to something like "Add missing platform information" and instead have an installation section that talks about `pip install pyorbital` or `conda install -c conda-forge pyorbital`?

Author: Good suggestion! Fixed, but tried keeping the installation part very short and simple. Okay?

Reviewer: I rewrote the instructions. I think it matches what I've written for other pytroll packages. I think it is important to mention the exact pip and conda commands for the usual use case. The installing from source commands are also important for the two cases of "install the dev version" cases for user support/bug fixes and the "you want to contribute, install in editable mode". I say that because I've heard users request these instructions over and over and over again.

Author: Looks fine, thx!
Realize that RTD does not build for PRs in this repo. Have forgot what I did for other projects to get this working.
Can we merge this then?

Reviewer: Looks like RTD builds have been failing for quite some time. Let me look into it.

Reviewer: I've enabled pull requests now too.

Reviewer: Ah it's been building on Python 2.7.

Reviewer: Looks good now. One of us can merge when the tests pass again.

This file already contain many low earth orbiting environmental or
meteorological satellites and thus likely be sufficient for your purpose.

But should it not contain your satellites of interest make a copy of the

> *platforms.txt* file and add the missing satellites and their NORAD identifiers and place

Author: Fixed

the file in the directory pointed to by :envvar:`PYORBITAL_CONFIG_PATH`.

The NORAD identifier can be found as the first number of each line in the
Two-Line Elements files (eg. from `celestrak`_).

> Pyorbital comes with a small script ``check_platform.py`` to check whether a

Reviewer: If this script is not part of the package then can this include a link to the script on github. If it is not part of the package then this sentence should be changed as "Pyorbital comes with" is not true.

If it does come with the package then `python -m pyorbital.check_platform ...` should be used instead of calling the script directly as that is how someone who has installed the package would access the script.

Author: This is done now

Installation
------------
Pyorbital comes with a file platforms.txt that maps a satellite name to the NORAD identifier.

> This file needs to be copied to the appropriate config directory ($PYORBITAL_CONFIG_PATH).

Author: Done. Added two sections at the bottom on the 2 env variables used

> This file needs to be copied to the appropriate config directory pointed to by :envvar:`PYORBITAL_CONFIG_PATH`.

Reviewer: These sentences make it seem like I *need* to create this `platforms.txt`, but the envvar description makes it seem optional. If it is required then these sentences here need to mention `platforms.txt` by name.

Author: I tried improving this, thanks!

    >>> tle.inclination
    99.043499999999995

If no path is provided pyorbital first tries to read any local TLE files in the
directory given by the environment variable :envvar:`TLES`. If this variable is not
set Pyorbital will try get the earth observation TLE files over the internet
from celestrak.com. Note this downloading only happens if no

> specific TLE file is provided or if the :envvar:`TLES` environment variable is not set.

Reviewer: Forgive the ignorance, but how does it search for other TLE files in this directory? What is the expected naming scheme? Or are TLEs known to have a specific naming scheme?

    >>> tle.inclination
    99.043499999999995

> If no path is provided pyorbital first tries to read any local tle files in the
> directory given by the environment variable TLES. If this environment is not
> set Pyorbital will try get the earth observation TLE-files over the internet
> from celestrak.com. Please observe, that it is only in the sace that no
> specific tle file is provided or if the TLES environment variable is not set
> that TLE fetch over the internet is attempted.

Reviewer: A couple typos and rewordings, but the phrase "environment variable TLES" is confusing. Is that a variable named "TLES" or are those TLEs found in the directory pointed to by the PYORBITAL_CONFIG_PATH? And is it "TLE" or "tle", it seems to be switched around a lot in the docs.

```suggestion
If no path is provided pyorbital first tries to read any local tle files in the
directory given by the environment variable TLES. If this environment is not
set Pyorbital will try get the earth observation TLE files over the internet
from celestrak.com. Note this downloading if no
specific tle file is provided or if the TLES environment variable is not set.
```

Author: > A couple typos and rewordings, but the phrase "environment variable TLES" is confusing. Is that a variable named "TLES" or are those TLEs found in the directory pointed to by the PYORBITAL_CONFIG_PATH? And is it "TLE" or "tle", it seems to be switched around a lot in the docs.

Yes, should be "TLE" rather than "tle". And, yes, TLES is an environment variable - the path to where the user keeps her/his local TLE files. Not sure if this env variable could actually be a list of directories. Need to check that. Do you remember @loerum ?

Reviewer: Ok then if TLES is an env var it should at least be highlighted with two backticks on either side, but I think you can use this in sphinx to make it extra clear:

https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-envvar

Author: Thanks, that is really nice. It works.

Author: Done

    >>> astronomy.sun_zenith_angle(utc_time, lon, lat)
    62.685986438071602

.. envvar:: PYORBITAL_CONFIG_PATH

   It is possible (but not mandatory) to define this environment variable to
   have full control of certain static data used by Pyorbital:

   Pyorbital comes with a file platforms.txt that maps a satellite name to the
   NORAD identifier. This internal file is accessed by Pyorbital without the
   user having to do anything.  But if you need to change or update this file
   you can make your own copy and place in the directory pointed to by this

>    environment variable.

Reviewer: Does the file still need to be called `platforms.txt`? Why is this a directory and not a single file path?

Author: Well, I think you are right that it could be a path to a single file, but way back I think we thought there would be more config files needed that could be accessed via this directory. Didn't want to change this now. We could make an issue perhaps?

> .. envvar:: TLES

Reviewer: I'd prefer if the first sentence in this description was what type of value this should have (is it a directory or a single file? does it accept multiple paths? if so, colon separated?) and if it is optional. Then I think the more detailed description can come after it.

Author: I think this should be better now?
